nodeup/pkg/model/cloudconfig.go

```diff
@@ -59,37 +66,39 @@ func (b *CloudConfigBuilder) Build(c *fi.ModelBuilderContext) error {
 		if cloudConfig.DisableSecurityGroupIngress != nil {
 			lines = append(lines, fmt.Sprintf("DisableSecurityGroupIngress = %t", *cloudConfig.DisableSecurityGroupIngress))
 		}
-	}
-
-	if cloudConfig.VSphereUsername != nil {
-		lines = append(lines, "user = "+*cloudConfig.VSphereUsername)
-	}
-
-	if cloudConfig.VSpherePassword != nil {
-		lines = append(lines, "password = "+*cloudConfig.VSpherePassword)
-	}
-
-	if cloudConfig.VSphereServer != nil {
-		lines = append(lines, "server = "+*cloudConfig.VSphereServer)
-		lines = append(lines, "port = 443")
-		lines = append(lines, fmt.Sprintf("insecure-flag = %t", true))
-	}
-
-	if cloudConfig.VSphereDatacenter != nil {
-		lines = append(lines, "datacenter = "+*cloudConfig.VSphereDatacenter)
-	}
-
-	if cloudConfig.VSphereDatastore != nil {
-		lines = append(lines, "datastore = "+*cloudConfig.VSphereDatastore)
+	case "vsphere":
+		vm_uuid, err := getVMUUID(b.Cluster.Spec.KubernetesVersion)
+		if err != nil {
+			return err
+		}
+		// Global Config for vSphere CloudProvider
+		if cloudConfig.VSphereUsername != nil {
+			lines = append(lines, "user = "+*cloudConfig.VSphereUsername)
+		}
+		if cloudConfig.VSpherePassword != nil {
+			lines = append(lines, "password = "+*cloudConfig.VSpherePassword)
+		}
+		if cloudConfig.VSphereServer != nil {
+			lines = append(lines, "server = "+*cloudConfig.VSphereServer)
+			lines = append(lines, "port = 443")
+			lines = append(lines, fmt.Sprintf("insecure-flag = %t", true))
+		}
+		if cloudConfig.VSphereDatacenter != nil {
+			lines = append(lines, "datacenter = "+*cloudConfig.VSphereDatacenter)
+		}
+		if cloudConfig.VSphereDatastore != nil {
+			lines = append(lines, "datastore = "+*cloudConfig.VSphereDatastore)
+		}
+		if vm_uuid != "" {
+			lines = append(lines, "vm-uuid = "+strings.Trim(vm_uuid, "\n"))
+		}
+		// Disk Config for vSphere CloudProvider
+		// We need this to support Kubernetes vSphere CloudProvider < v1.5.3
+		lines = append(lines, "[disk]\n"+"scsicontrollertype = pvscsi")
```

Comment:
We need ```global``` before ```disk``` in vsphere conf file. This will do the same. Earlier, disk configuration was added to conf var. Now, we directly append to lines var. 

I had plan to include network config in this PR. Will create a separate PR